extensions/blocks/map/map.php

```diff
@@ -79,6 +93,8 @@ function jetpack_map_block_load_assets( $attr, $content ) {
 		);
 	}
 
+	jetpack_record_mapbox_wpcom_load_event();
```

Comment:
I believe this will double-count Map blocks in AMP requests. Here's what I'd recommend, which will cause the tracking function to fire once per Map block regardless of AMP status:

1) In `jetpack_record_mapbox_wpcom_load_event()`, bail if the AMP query params are present, like so:

```
function jetpack_record_mapbox_wpcom_load_event() {

	$map_block_counter = isset( $_GET, $_GET['map-block-counter'] ) ? absint( $_GET['map-block-counter'] ) : null;
	// phpcs:ignore WordPress.Security.NonceVerification
	$map_block_post_id = isset( $_GET, $_GET['map-block-post-id'] ) ? absint( $_GET['map-block-post-id'] ) : null;

	if ( $map_block_counter && $map_block_post_id ) {
		return;
	}
```

2) In `jetpack_map_block_load_assets()` move `jetpack_record_mapbox_wpcom_load_event();` above the AMP check.

3) Remove entirely the second call to `jetpack_record_mapbox_wpcom_load_event()`.

Reply:
@jeffersonrabb AFAICS, in AMP requests, we wouldn't get to this: https://github.com/Automattic/jetpack/blob/8f0098b3e3e3beb622c1f37bb7df1cb6f381dd29/extensions/blocks/map/map.php#L96
Because we early return here: https://github.com/Automattic/jetpack/blob/8f0098b3e3e3beb622c1f37bb7df1cb6f381dd29/extensions/blocks/map/map.php#L87-L92

But yeah, reading again the code, I guess that moving the recording above the AMP check  (and removing the second call) should be safe enough in all cases.